ledger.go

```diff
@@ -802,7 +810,9 @@ func (l *Ledger) query() {
 				vote.block = &block
 			}
 			l.metrics.queryLatency.Time(f)
-		}(p)
+		}
+
+		l.queueWorkerPool.Queue(func() { f(p) })
```

Comment:
CI caught a data race somewhere here.

Reply:
I assume parameter enclosure for embeded function works differently